### DISCUSSION TITLE
Dynamic Pydantic schemas based on user

### DISCUSSION THREAD

**Discussion by alispa** [fastapi/fastapi#5252] [Questions] [answered]
```
@router.post("/example")
def create_something(data: UsersPydanticSchema):
      return "OK"
```

Is there a way to have `UsersPydanticSchema `customized for every user, for example place them into folders like `user_A/schemas.py -> UsersPydanticSchema` , or maybe there is a better idea.

Then, on every request, we get the user's folder name (ex. `user_A`) from the `session` or the `token` and call the proper `UsersPydanticSchema` .

**Comment by YuriiMotov** [comment, marked as answer]
The only way is to define a universal model that will fit for all users. Maybe using discriminated unions.

Or, if you don't care about openapi schema, you can exclude body parameters from endpoint declaration and validate body data manually from `request.json`

```python
@app.post("/submit")
async def submit(request: Request):
    user_id = request.headers.get("X-User-ID")
    if not user_id:
        raise HTTPException(status_code=400, detail="Missing user ID")

    schema_cls = get_user_schema(user_id)

    try:
        body = await request.json()
        validated_data = schema_cls(**body)
    except ValidationError as e:
        raise HTTPException(status_code=422, detail=e.errors())
   
```
